modules/apps/portal/portal-http-impl/src/test/java/com/liferay/portal/http/internal/HttpImplTest.java

```diff
@@ -63,6 +144,98 @@ public void testIsNonProxyHost() throws Exception {
 		}
 	}
 
+	private Tuple _getHttpConnectionStrategies() {
+		ClientExecChain clientExecChain = ReflectionTestUtil.getFieldValue(
+			(CloseableHttpClient)ReflectionTestUtil.invoke(
+				_httpImpl, "getCloseableHttpClient",
+				new Class<?>[] {HttpHost.class}, new Object[] {null}),
+			"execChain");
+
+		while (true) {
+			clientExecChain = ReflectionTestUtil.getFieldValue(
+				clientExecChain, "requestExecutor");
+
+			if (clientExecChain instanceof MainClientExec) {
+				return new Tuple(
+					ReflectionTestUtil.getFieldValue(
+						clientExecChain, "keepAliveStrategy"),
+					ReflectionTestUtil.getFieldValue(
+						clientExecChain, "reuseStrategy"));
+			}
+		}
+	}
+
+	private void _setHttpKeepAliveTimeout(int keepAliveTimeout) {
+		_httpConfigurationProperties.put("keepAliveTimeout", keepAliveTimeout);
+	}
+
+	private void _testHttpKeepAlive(
+		boolean expectedKeepAlive,
+		long expectedKeepAliveTimeoutInMilliseconds) {
+
+		_httpImpl.activate(_httpConfigurationProperties);
+
+		Tuple connectionStrategiesTuple = _getHttpConnectionStrategies();
+
+		ConnectionReuseStrategy connectionReuseStrategy =
+			(ConnectionReuseStrategy)connectionStrategiesTuple.getObject(1);
+
+		Assert.assertEquals(
+			expectedKeepAlive,
+			connectionReuseStrategy.keepAlive(_httpResponse, _httpContext));
+
+		long keepAliveTimeout = -1;
+
+		if (expectedKeepAlive) {
+			ConnectionKeepAliveStrategy connectionKeepAliveStrategy =
+				(ConnectionKeepAliveStrategy)
+					connectionStrategiesTuple.getObject(0);
+
+			BasicPoolEntry basicPoolEntry = new BasicPoolEntry(
+				"id", new HttpHost("localhost", 8080),
+				new DefaultManagedHttpClientConnection("id", 8 * 1024));
+
+			basicPoolEntry.updateExpiry(
+				connectionKeepAliveStrategy.getKeepAliveDuration(
+					_httpResponse, new BasicHttpContext(null)),
+				TimeUnit.MILLISECONDS);
+
+			keepAliveTimeout = basicPoolEntry.getExpiry();
+
+			if (keepAliveTimeout != Long.MAX_VALUE) {
+				keepAliveTimeout -= basicPoolEntry.getUpdated();
+			}
+		}
+
+		Assert.assertEquals(
+			expectedKeepAliveTimeoutInMilliseconds, keepAliveTimeout);
+	}
+
+	private void _testHttpKeepAlive(
+		boolean expectedKeepAlive, long expectedKeepAliveTimeoutInMilliseconds,
+		long keepAliveTimeoutHeaderValue) {
+
+		_httpResponse.setHeaders(
+			new Header[] {
+				new BasicHeader(HttpHeaders.CONNECTION, HttpHeaders.KEEP_ALIVE),
+				new BasicHeader(HttpHeaders.CONTENT_LENGTH, "10")
+			});
+
+		if (keepAliveTimeoutHeaderValue > -1) {
+			_httpResponse.setHeader(
+				HttpHeaders.KEEP_ALIVE,
+				"timeout=" + keepAliveTimeoutHeaderValue);
+		}
+
+		_testHttpKeepAlive(
+			expectedKeepAlive, expectedKeepAliveTimeoutInMilliseconds);
+	}
+
+	private final Map<String, Object> _httpConfigurationProperties =
```

Comment:
@ericyanLr Please remove this map, it is only used to store "keepAliveTimeout", you can just pass the value of "keepAliveTimeout" as a parameter